### PR TITLE
Feature/add table splitting kvs tests

### PR DESCRIPTION
**Pull request by joelea**
Adds some tests for TableSplittingKeyValueService and moves around some code inside. One section was just a bit tricky to understand, the other made intellij sad about the duplication.

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java [hunk 1]
```diff
@@ -89,6 +92,13 @@ public void createTable(String tableName, byte[] tableMetadata) {
 
     @Override
     public void createTables(Map<String, byte[]> tableNamesToTableMetadata) {
+        Map<KeyValueService, Map<String, byte[]>> splitTableNamesToTableMetadata = groupByDelegate(tableNamesToTableMetadata);
+        for (KeyValueService delegate : splitTableNamesToTableMetadata.keySet()) {
```
Comment (clockfort) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java]: this case doesn't matter for perf at all and you don't have to change it, but just for SA at least my IDE would yell at me if I wrote this, because using the entrySet here instead is more efficient since you're also doing the get(key) inside here

Comment (joelea) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TableSplittingKeyValueService.java]: This was already in there. Happy to change it to the entry set version, but didn't want to go around making things less clear for what looks like an irrelevant perf gain.

**Comment by tmgordeeva** [comment]
Unrelated to your change, but after reviewing it looks like getMetadataForTables here is wrong. It gets all the metadata for each delegate regardless of the table split, so it may return some other KVSs possibly outdated metadata instead of the proper delegate. It should get the same treatment as getAllTableNames. Not related so could just be a separate change.

**Comment by joelea** [comment]
That sounds sucky - but the kind of thing I thought we'd find when I started writing the tests for it. I'll do another pass on the methods I don't have good tests for yet as a separate change and flag anything weird as well as fixing the bit that you just mentioned.

**Comment by clockfort** [comment]
I'm not sure it's actually wrong, though I see where tanya is going with talking about tables with same names that used to exist in different places.
I think we should already have in place the contract that says you can't have same table name in two places and that dropping a table synchronously removes its metadata, which would make this simple implementation of getMetadataForTables valid.

**Comment by tmgordeeva** [comment]
I mean it's sort of valid now - if the split KVS had older metadata from some previous hydration before the table split was set up, it could still be wrong. We're supposed to support updating table metadata without dropping the table first. @clockfort don't see a contract that you can't have a table name in two places or at least any enforcement of it globally throughout all KVS calls.
